scripts/ci/lint-clang.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1091
+#
+# Reformats C/C++ code belonging to the repository.
+# The primary Linter with support for reformatting
+# files is clang-format. Any reformatted code causes
+# non-zero exit code (CI failure).
+# Additionally, there can be secondary Linters
+# used to raise warnings.
+
+REPOROOT=$(realpath "$(dirname "$0")/../..")
+PATTERN="./*.h ./*.cpp ./*.c"
+VERSION=16
+for V in $(seq $((VERSION+10)) -1 ${VERSION}); do
+  LINTER=$(command -v "clang-format-${V}")
+  if [ "${LINTER}" ]; then break; fi
+done
+if [ ! "${LINTER}" ]; then
+  LINTER=$(command -v clang-format)
+fi
+
+if [ "${LINTER}" ]; then
```

Comment:
Early exit:
```
  if [ ! "$LINTER" ]; then
    echo "WARNING: missing C/C++-linter (${LINTER})."
    exit 1
  fi
  ...
```

Reply:
Hmm, now I realized why it is nested. Idea was to treat missing `clang-format` as well as missing `black` like a "soft-error". Means exit code is zero but no check/reformat is done. This may help us to get things running with less requirements for the target system (like installing additional Python packages).